pypeit/extraction.py

```diff
@@ -247,6 +245,18 @@ def __init__(self, sciImg, slits, sobjs_obj, spectrograph, par, objtype, global_
         else:
             msgs.warn("Spectral FWHM image could not be generated")
 
+        # get flatfield image for blaze function
+        self.flatimg = None
+        if flatimages is not None:
+            # This way of getting the flat image (instead of just reading flatimages.pixelflat_model) ensures that the
+            # flat image is available also when an archival pixel flat is used.
+            flat_raw = flatimages.pixelflat_raw if flatimages.pixelflat_raw is not None else flatimages.illumflat_raw
+            if flat_raw is not None and flatimages.pixelflat_norm is not None:
+                # TODO: Can we just use flat_raw if flatimages.pixelflat_norm is None?
```

Comment:
I think we can, but perhaps add a warning that the raw data will be used?

Reply:
Thanks for your input. I'll leave it as a TODO for now.